virttest/libvirt_xml/base.py

```diff
@@ -320,6 +320,70 @@ def setup_attrs(self, **attrs):
                 target_obj.setup_attrs(**value)
                 setattr(self, key, target_obj)
 
+    def fetch_attrs(self):
+        """
+        Fetch attributes of xml object
+
+        :return: dict-type attributes, keys were pre-defined in xml classes
+
+        Example:
+        A dimm device 'dimm_xml' has xml like this:
+            <memory model="dimm">
+                <target>
+                    <size unit="KiB">524288</size><node>0</node>
+                </target>
+                <alias name="dimm2" />
+                <address base="0x120000000" slot="2" type="dimm" />
+            </memory>
+
+        dimm_xml.fetch_attrs()
+
+        Output should be:
+            {
+                'target': {
+                    'node': 0, 'size': 524288, 'size_unit': 'KiB'
+                    },
+                'alias': {'name': 'dimm2'},
+                'address': {
+                    'attrs': {
+                        'base': '0x120000000', 'slot': '2', 'type': 'dimm'
+                        },
+                    'type_name': 'dimm'
+                    },
+                'mem_model': 'dimm'
+            }
+
+        """
+        attrs = {}
+        slots = set(self.__all_slots__) - set(self.__uncompareable__) - {'device_tag'}
+        for key in slots:
+            try:
+                # Try to get values of each attribute, slot by slot
+                value = self[key]
+            except (xcepts.LibvirtXMLAccessorError,
+                    xcepts.LibvirtXMLNotFoundError,
+                    AttributeError,
+                    KeyError):
+                continue
```

Comment:
Is 'pass' better than 'continue' here because there is no other codes to be executed after this line in the loop and the loop will continue by default?

Reply:
After discussing offline, I understand the origin purpose, so I accept we use 'continue' instead of 'pass' to forcefully repeat the loop without caring about its following logic.